lib/config_cat.ex

```diff
@@ -0,0 +1,187 @@
+
+  def start_link(nil, _options), do: {:error, :missing_sdk_key}
+
+  def start_link(sdk_key, options) do
+    with {name, options} <- Keyword.pop(options, :name, __MODULE__),
+         {initial_config, options} <- Keyword.pop(options, :initial_config) do
+      initial_state = %{
+        config: initial_config,
+        etag: nil,
+        last_update: nil,
+        options: Keyword.merge(default_options(), options),
+        sdk_key: sdk_key
+      }
+
+      GenServer.start_link(__MODULE__, initial_state, name: name)
+    end
+  end
+
+  defp default_options, do: [api: ConfigCat.API, fetch_policy: FetchPolicy.auto()]
+
+  def get_value(key, default_value, user_or_options \\ []) do
+    if Keyword.keyword?(user_or_options) do
+      get_value(key, default_value, nil, user_or_options)
+    else
+      get_value(key, default_value, user_or_options, [])
+    end
+  end
+
+  def get_value(key, default_value, user, options) do
+    client = Keyword.get(options, :client, __MODULE__)
+    GenServer.call(client, {:get_value, key, default_value, user})
+  end
+
+  def force_refresh(client \\ __MODULE__) do
+    GenServer.call(client, :force_refresh)
+  end
+
+  @impl GenServer
+  def init(state) do
+    {:ok, state, {:continue, :maybe_init_fetch}}
+  end
+
+  @impl GenServer
+  def handle_call({:get_value, key, default_value, user}, _from, state) do
+    with {:ok, new_state} <- maybe_refresh(state),
+         value <- Rollout.evaluate(key, user, default_value, new_state.config) do
+      {:reply, value, new_state}
+    else
+      error -> {:reply, error, state}
+    end
+  end
+
+  @impl GenServer
+  def handle_call(:force_refresh, _from, state) do
+    with {:ok, new_state} <- refresh(state) do
+      {:reply, :ok, new_state}
+    else
+      error -> {:reply, error, state}
+    end
+  end
+
+  defp schedule_initial_fetch?(%{options: options}) do
+    options
+    |> Keyword.get(:fetch_policy)
+    |> FetchPolicy.schedule_initial_fetch?()
+  end
+
+  defp maybe_refresh(%{options: options} = state) do
+    options
+    |> Keyword.get(:fetch_policy)
+    |> maybe_refresh(state)
+  end
+
+  defp maybe_refresh(fetch_policy, %{last_update: last_update} = state) do
+    if FetchPolicy.needs_fetch?(fetch_policy, last_update) do
+      refresh(state)
+    else
+      {:ok, state}
+    end
+  end
+
+  defp refresh(%{options: options, etag: etag} = state) do
+    Logger.info("Fetching configuration from ConfigCat")
+
+    with api <- Keyword.get(options, :api),
+         {:ok, response} <- api.get(url(state), headers(etag)) do
+      response
+      |> log_response()
+      |> handle_response(state)
+    else
+      error ->
+        log_error(error)
+    end
+  end
+
+  defp handle_response(%Response{status_code: code, body: body, headers: headers}, state)
+       when code >= 200 and code < 300 do
+    with {:ok, config} = Jason.decode(body),
+         etag <- extract_etag(headers) do
+      {:ok, %{state | config: config, etag: etag, last_update: now()}}
```

Comment:
And here comes the magic to handle our new datagovernance feature. Please adjust this with the same logic from here:
https://github.com/configcat/python-sdk/blob/master/configcatclient/configfetcher.py#L74

Reply:
Extracted to #13